### DISCUSSION TITLE
How does the connection work exactly ?

### DISCUSSION THREAD

**Discussion by uaygunn** [gravitational/teleport#20907] [Q&A] [answered]
Hi guys i have a question about connection after something weird happened

I was testing the conenction btw teleport server and a host. ( host 10.34.102.30 and server is 10.34.102.15)

I blocked all connections from server to host in firewall. (from host firewall - ubuntu) afterthat i tried ssh to it from server terminal but the connection was refused as should be. But when i try from web gui i could connect to my host without problem.

So my question is does teleport use clients network and access to host or it directly connects from server ?

I ask this because i am curious if i can connect different servers from different locations (and of course different networks) without giving access to server ( because all networks are different and i need to create ipsec to all firewalls and i am not sure that my customers will allow me to do)

To short can i connect other hosts from my computer if i am in a vpn connection in host's network but my server is not ?

**Comment by webvictim** [comment, marked as answer]
There are three potential answers to this question.

If your node is joined directly to the Teleport auth service (i.e. it has `auth_server: teleport.example.com:3025` or similar in its config file), then:
**Answer 1**: if you run `tsh ssh user@hostname` from your local machine, you are making a connection from your local machine's IP to port 3022 on the host
**Answer 2**: if you connect to the host using the Teleport web UI, you are making a connection from the IP of the machine where the Teleport proxy is running to port 3022 on the host

If your node is joined to the proxy service using IoT mode/node tunnelling (i.e. it has `proxy_server: teleport.example.com:443` in its config file), then:
**Answer 3**: either running `tsh ssh user@hostname` or using the Teleport web UI will use the active reverse tunnel from the node to the Teleport proxy to make the connection. In this situation, the only outbound connection from your machine is to the Teleport proxy service and the IP of the machine is irrelevant.

For your use case, I would recommend that you join your nodes to the proxy service using reverse tunnels. This means you don't need VPN connections from yourself to all the servers - they all connect to the Teleport proxy service, and you connect to that too.

I hope this is clear.

**Reply by uaygunn** [reply]
Yeah i understood that thank you for your answers :) So to short teleport server should be able to access hosts in this situation.

I think that my customers will not allow me to tunnel btw my network ( which will teleport running) to their network ( which hosts that i want to connect via teleport running ). 

Thank you again

**Reply by webvictim** [reply]
In general - as the connection is an outbound tunnel from their network to yours, it can be considered more secure than opening up firewall rules to allow access to the nodes from elsewhere. The tunnels could theoretically work over a site-to-site VPN as well if needed.
